source/dxgi/dxgi_factory.cpp

```diff
@@ -0,0 +1,849 @@
+	reshade::log::message(reshade::log::level::debug, "Destroying IDXGIFactory%hu object %p (%p).", interface_version, this, orig);
+#endif
+	delete this;
+
+	{
+		// Only release internal reference after the effect runtime has been destroyed, so any references it held are cleaned up at this point
+		const ULONG ref_orig = orig->Release();
+		if (ref_orig != 0) // Verify internal reference count
+			reshade::log::message(reshade::log::level::warning, "Reference count for IDXGIFactory%hu object %p (%p) is inconsistent (%lu).", interface_version, this, orig, ref_orig);
+	}
+
+	return 0;
+}
+
+HRESULT STDMETHODCALLTYPE DXGIFactory::SetPrivateData(REFGUID Name, UINT DataSize, const void *pData)
+{
+	return _orig->SetPrivateData(Name, DataSize, pData);
+}
+HRESULT STDMETHODCALLTYPE DXGIFactory::SetPrivateDataInterface(REFGUID Name, const IUnknown *pUnknown)
+{
+	return _orig->SetPrivateDataInterface(Name, pUnknown);
+}
+HRESULT STDMETHODCALLTYPE DXGIFactory::GetPrivateData(REFGUID Name, UINT *pDataSize, void *pData)
+{
+	return _orig->GetPrivateData(Name, pDataSize, pData);
+}
+HRESULT STDMETHODCALLTYPE DXGIFactory::GetParent(REFIID riid, void **ppParent)
+{
+	return _orig->GetParent(riid, ppParent);
+}
+
+HRESULT STDMETHODCALLTYPE DXGIFactory::EnumAdapters(UINT Adapter, IDXGIAdapter **ppAdapter)
+{
+	return _orig->EnumAdapters(Adapter, ppAdapter);
+}
+
+HRESULT STDMETHODCALLTYPE DXGIFactory::MakeWindowAssociation(HWND WindowHandle, UINT Flags)
+{
+	return _orig->MakeWindowAssociation(WindowHandle, Flags);
+}
+
+HRESULT STDMETHODCALLTYPE DXGIFactory::GetWindowAssociation(HWND *pWindowHandle)
+{
+	return _orig->GetWindowAssociation(pWindowHandle);
+}
+
+HRESULT STDMETHODCALLTYPE DXGIFactory::CreateSwapChain(IUnknown *pDevice, DXGI_SWAP_CHAIN_DESC *pDesc, IDXGISwapChain **ppSwapChain)
+{
+
+	if (g_in_dxgi_runtime)
+		return _orig->CreateSwapChain(pDevice, pDesc, ppSwapChain);
+
+	reshade::log::message(
+		reshade::log::level::info,
+		"Redirecting IDXGIFactory::CreateSwapChain(this = %p, pDevice = %p, pDesc = %p, ppSwapChain = %p) ...",
+		_orig, pDevice, pDesc, ppSwapChain);
+
+	if (pDevice == nullptr || pDesc == nullptr || ppSwapChain == nullptr)
+		return DXGI_ERROR_INVALID_CALL;
+
+	com_ptr<IUnknown> device_proxy;
+	const reshade::api::device_api direct3d_version = query_device(pDevice, device_proxy);
+
+	DXGI_SWAP_CHAIN_DESC desc = *pDesc;
+	UINT sync_interval = UINT_MAX;
+	const bool modified = dump_and_modify_swapchain_desc(direct3d_version, desc, sync_interval);
+
+	g_in_dxgi_runtime = true;
+	const HRESULT hr = _orig->CreateSwapChain(pDevice, &desc, ppSwapChain);
+	g_in_dxgi_runtime = false;
+	if (FAILED(hr))
+	{
+		reshade::log::message(reshade::log::level::warning, "IDXGIFactory::CreateSwapChain failed with error code %s.", reshade::log::hr_to_string(hr).c_str());
+		return hr;
+	}
+
+	DXGI_SWAP_CHAIN_DESC orig_desc = *pDesc;
+	if (!modified)
+		orig_desc.BufferCount = 0;
+	orig_desc.BufferDesc.Format = desc.BufferDesc.Format;
```

Comment:
This line is new, is it intentional (on first glance seems weird to return the modified format, rather than the original, when this was originally designed to return the original description)?

Reply:
You're right, must have been from testing.